share/completions/mkdir.fish

```diff
@@ -0,0 +1,9 @@
+complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x
+complete -c mkdir -s p -l parents --description 'Make parent directories as needed'
+complete -c mkdir -s v -l verbose --description 'Print a message for each created directory'
+complete -c mkdir -s Z --description 'Set SELinux security context of each created directory to the default type'
```

Comment:
Seems solaris does:

http://docs.oracle.com/cd/E23824_01/html/821-1462/modinfo-1m.html

But if you can only manage to  provide completions relevant to a Linux modinfo, it seems reasonable to check uname.